system/lib/wasmfs/file.cpp

```diff
@@ -25,6 +25,68 @@ std::shared_ptr<File> Directory::Handle::getEntry(std::string pathName) {
 //
 // Path Parsing utilities
 //
+ParsedPath getParsedPath(std::vector<std::string> pathParts,
+                         long& err,
+                         std::shared_ptr<File> forbiddenAncestor) {
+  std::shared_ptr<Directory> curr;
+  auto begin = pathParts.begin();
+
+  if (pathParts.empty()) {
+    err = -ENOENT;
+    return ParsedPath{{}, nullptr};
+  }
+
+  // Check if the first path element is '/', indicating an absolute path.
+  if (pathParts[0] == "/") {
+    curr = wasmFS.getRootDirectory();
+    begin++;
+    // If the pathname is the root directory, return the root as the child.
+    if (pathParts.size() == 1) {
+      return ParsedPath{curr->locked(), curr};
+    }
+  } else {
+    curr = wasmFS.getCWD();
+  }
+
+  for (auto pathPart = begin; pathPart != pathParts.end() - 1; ++pathPart) {
+    // Find the next entry in the current directory entry
+#ifdef WASMFS_DEBUG
+    directory->locked().printKeys();
+#endif
+    auto entry = curr->locked().getEntry(*pathPart);
+
+    if (forbiddenAncestor) {
+      if (entry == forbiddenAncestor) {
+        err = -EINVAL;
+        return ParsedPath{{}, nullptr};
+      }
+    }
+
+    // Requested entry (file or directory)
```

Comment:
It looks like this comment is explaining what `entry` is, but then I would expect it to be up where `entry` is defined.